lib/heap_sort.rb

```diff
@@ -1,8 +1,19 @@
+# This method uses a heap to sort an array.
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 
+def heapsort(list)
```

Comment:
👍 Adding an element to a heap takes O(log n) time and you're doing it n times... so the time complexity is O(n log n)